File: backend/src/SyncChat.Application/Models/Message.cs
using System;

namespace SyncChat.Application.Models;

public class Message
{
    public string Id { get; set; } = string.Empty;
    public string ConversationId { get; set; } = string.Empty;
    public string SenderId { get; set; } = string.Empty;
    public string Text { get; set; } = string.Empty;
    public string? MediaUrl { get; set; }
    public DateTime Timestamp { get; set; }
    public string[] ReadBy { get; set; } = Array.Empty<string>();
}

File: frontend/app/src/main/java/com/syncchat/app/data/model/Message.kt
package com.syncchat.app.data.model

import java.util.Date

data class Message(
    val id: String = "",
    val conversationId: String = "",
    val senderId: String = "",
    val text: String = "",
    val mediaUrl: String? = null,
    val timestamp: Date = Date(),
    val readBy: List<String> = emptyList(),
    val status: String = MessageStatus.SENT.name
)

enum class MessageStatus {
    PENDING,   // Stored locally, not yet sent to server
    SENT,      // Successfully delivered to backend
    FAILED     // Delivery attempted but failed (will retry)
}
